arch/common/gen_isr_tables.py

```diff
@@ -182,7 +228,40 @@ def main():
                 error("Regular Interrupt %d declared with parameter 0x%x "
                         "but no SW ISR_TABLE in use"
                         % (irq, param))
-            swt[irq - offset] = (param, func)
+
+            if not "CONFIG_MULTI_LEVEL_INTERRUPTS" in syms:
+                swt[irq - offset] = (param, func)
+            else:
+                # Figure out third level interrupt position
+                debug('IRQ = ' + hex(irq))
+                irq3 = (irq & 0x00FF0000) >> 16
```

Comment:
what is this magic number? is it HW independent? Can you define it somewhere and explain what it is?

Reply:
The magic number is hardware independent. It represents the various rungs of interrupts. 0x000000FF represents the first rung (i.e. the interrupts that directly go to the processor), 0x0000FF00 represents the second rung (i.e. the interrupts funnel into one line which then goes into the first rung) and so on. Since this is a python file, i didn't know where to put the magic number (or #define it).